timelock-impl/src/main/java/com/palantir/atlasdb/timelock/clock/ClockSkewMonitor.java

```diff
@@ -76,35 +76,48 @@ public static ClockSkewMonitor create(Set<String> remoteServers, Optional<SSLSoc
     }
 
     public void runInBackground() {
-        executorService.scheduleAtFixedRate(this::runOnce, 0, PAUSE_BETWEEN_REQUESTS.toNanos(), TimeUnit.NANOSECONDS);
+        executorService.scheduleWithFixedDelay(
+                this::runOnce, 0, PAUSE_BETWEEN_REQUESTS.toNanos(), TimeUnit.NANOSECONDS);
     }
 
     private void runOnce() {
-        try {
-            runInternal();
-        } catch (Throwable t) {
-            events.exception(t);
-        }
+        Map<String, RequestTime> newRequests = getRemoteRequestTimes();
+        checkAndUpdatePreviousRequestTimes(newRequests);
     }
 
-    private void runInternal() {
-        clocksByServer.forEach((server, remoteClockService) -> {
-            long localTimeAtStart = localClockService.getSystemTimeInNanos();
-            long remoteSystemTime = remoteClockService.getSystemTimeInNanos();
-            long localTimeAtEnd = localClockService.getSystemTimeInNanos();
+    private Map<String, RequestTime> getRemoteRequestTimes() {
+        Map<String, RequestTime> newRequestTimes = Maps.newHashMap();
 
-            RequestTime previousRequest = previousRequestsByServer.get(server);
-            RequestTime newRequest = RequestTime.builder()
-                    .localTimeAtStart(localTimeAtStart)
-                    .localTimeAtEnd(localTimeAtEnd)
-                    .remoteSystemTime(remoteSystemTime)
-                    .build();
+        clocksByServer.forEach((host, clockService) -> {
```

Comment:
don't want to pile too much onto this PR, but thinking about this, it's probably better to schedule each server on its own thread so slow servers dont block everyone else (e.g. if they are down and we take a while to get a connect timeout)

Reply:
I'll open up a follow-up that shifts these to their own thread. The idea makes sense, definitely, but would like to get this in soon as the log-spam is pretty annoying!